### PR TITLE
misc(Downsampler): register metric reporters for downsampler spark jobs

### PR DESCRIPTION
**Pull request by sherali42**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?

- Register Kamon plugged-in metric reporters for Downsampled spark jobs
- Remove spark-context stop statement from the Driver

### CODE REVIEW

File: spark-jobs/src/main/scala/filodb/downsampler/index/DSIndexJob.scala [hunk 1]
```diff
@@ -20,12 +21,14 @@ object DSIndexJob extends StrictLogging with Instance {
...
   val dsJobsettings = DownsamplerSettings
 
-  private val kamonTags = Map( "rawDataset" -> dsJobsettings.rawDatasetName,
-    "owner" -> "BatchIndexUpdater")
-
   private val readSched = Scheduler.io("cass-index-read-sched")
   private val writeSched = Scheduler.io("cass-index-write-sched")
 
+  val taskCounter = Kamon.counter("num-tasks")
+  val completionCounter = Kamon.counter("num-completed-tasks")
```
Comment (vishramachandran) [inline spark-jobs/src/main/scala/filodb/downsampler/index/DSIndexJob.scala]: rename to . sparkForeachTasksCompleted ?

File: spark-jobs/src/main/scala/filodb/downsampler/index/DSIndexJob.scala [hunk 2]
```diff
@@ -20,12 +21,14 @@ object DSIndexJob extends StrictLogging with Instance {
   val settings = DownsamplerSettings
   val dsJobsettings = DownsamplerSettings
 
-  private val kamonTags = Map( "rawDataset" -> dsJobsettings.rawDatasetName,
-    "owner" -> "BatchIndexUpdater")
-
   private val readSched = Scheduler.io("cass-index-read-sched")
   private val writeSched = Scheduler.io("cass-index-write-sched")
 
+  val taskCounter = Kamon.counter("num-tasks")
```
Comment (vishramachandran) [inline spark-jobs/src/main/scala/filodb/downsampler/index/DSIndexJob.scala]: Rename to `sparkTasksStarted` ? metric name too,

File: spark-jobs/src/main/scala/filodb/downsampler/index/DSIndexJob.scala [hunk 3]
```diff
@@ -55,25 +58,36 @@ object DSIndexJob extends StrictLogging with Instance {
   def updateDSPartKeyIndex(shard: Int, epochHour: Long): Unit = {
     import DSIndexJobSettings._
 
+    taskCounter.increment
+
+    val span = Kamon.buildSpan("timetaken-index-migration")
+      .withTag("shard", shard)
+      .withTag("epochHour", epochHour)
```
Comment (vishramachandran) [inline spark-jobs/src/main/scala/filodb/downsampler/index/DSIndexJob.scala]: epochHour becomes a high-card tag. Lets drop.
Also querying will result in too many time series

File: spark-jobs/src/main/scala/filodb/downsampler/index/DSIndexJob.scala (hunk 2 above)
Comment (sherali42) [reply spark-jobs/src/main/scala/filodb/downsampler/index/DSIndexJob.scala]: updated variable/metric names

File: spark-jobs/src/main/scala/filodb/downsampler/index/DSIndexJob.scala (hunk 3 above)
Comment (sherali42) [reply spark-jobs/src/main/scala/filodb/downsampler/index/DSIndexJob.scala]: removed epochHour tag

File: spark-jobs/src/main/scala/filodb/downsampler/index/DSIndexJob.scala (hunk 1 above)
Comment (sherali42) [reply spark-jobs/src/main/scala/filodb/downsampler/index/DSIndexJob.scala]: done

**Pull request merged**